src/pages/api/dataset-search.ts

```diff
@@ -5,6 +5,13 @@ export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse
 ) {
-  const searchResult = await datasetFetcher.search();
+  if (req.method !== 'POST') {
```

Comment:
I chose POST instead of GET because I had concerns about the URL size, because of the long resource ID and the possibility to check multiple filters. But maybe this is not a valid concern. Because this is a POST for getting resources, it is not RESTful, so I choose a non-RESTful name (other question). Let me check the GET once more.